test/unit/index.test.ts

```diff
@@ -0,0 +1,161 @@
+  ['AutoEncryptionLoggerLevel',          'reason'],
+  ['MongoErrorLabel',                    'reason'],
+  ['ExplainVerbosity',                   'reason'],
+  ['LoggerLevel',                        'reason'],
+  ['ServerApiVersion',                   'reason'],
+  ['BSONType',                           'reason'],
+  ['ReturnDocument',                     'reason'],
+  ['ProfilingLevel',                     'reason'],
+  ['ReadConcernLevel',                   'reason'],
+  ['ReadPreferenceMode',                 'reason'],
+  ['ServerType',                         'reason'],
+  ['TopologyType',                       'reason'],
+  ['ReadConcern',                        'reason'],
+  ['ReadPreference',                     'reason'],
+  ['WriteConcern',                       'reason'],
+  // events
+  ['CommandFailedEvent',                 'reason'],
+  ['CommandStartedEvent',                'reason'],
+  ['CommandSucceededEvent',              'reason'],
+  ['ConnectionCheckedInEvent',           'reason'],
+  ['ConnectionCheckedOutEvent',          'reason'],
+  ['ConnectionCheckOutFailedEvent',      'reason'],
+  ['ConnectionCheckOutStartedEvent',     'reason'],
+  ['ConnectionClosedEvent',              'reason'],
+  ['ConnectionCreatedEvent',             'reason'],
+  ['ConnectionPoolClearedEvent',         'reason'],
+  ['ConnectionPoolClosedEvent',          'reason'],
+  ['ConnectionPoolCreatedEvent',         'reason'],
+  ['ConnectionPoolMonitoringEvent',      'reason'],
+  ['ConnectionReadyEvent',               'reason'],
+  ['ServerClosedEvent',                  'reason'],
+  ['ServerDescriptionChangedEvent',      'reason'],
+  ['ServerHeartbeatFailedEvent',         'reason'],
+  ['ServerHeartbeatStartedEvent',        'reason'],
+  ['ServerHeartbeatSucceededEvent',      'reason'],
+  ['ServerOpeningEvent',                 'reason'],
+  ['TopologyClosedEvent',                'reason'],
+  ['TopologyDescriptionChangedEvent',    'reason'],
+  ['TopologyOpeningEvent',               'reason'],
+  ['SrvPollingEvent',                    'reason'],
+
+  ['GridFSBucketReadStream',             'reason'],
+  ['GridFSBucketWriteStream',            'reason'],
+  ['OrderedBulkOperation',               'reason'],
+  ['UnorderedBulkOperation',             'reason'],
+
+  // TS-NODE Adds these keys but they are undefined... weird
+  ['AnyBulkWriteOperation',              'ts-node'],
+  ['BulkWriteOptions',                   'ts-node'],
+]);
+
+describe('mongodb entrypoint', () => {
+  it('should export all keys in our exports list', () => {
+    expect(sorted(Object.keys(mongodb), byStrings)).to.deep.equal(
+      sorted(expectedExports.keys(), byStrings)
+    );
+  });
+
+  it('should export only keys in our exports list', () => {
+    const currentExports = Object.keys(mongodb);
+    const difference = setDifference(currentExports, expectedExports.keys());
+    expect(
+      difference,
+      `Found extra exports [${Array.from(difference).join(
```

Comment:
I think this doesn't differentiate between extra exports vs missing exports, do we want to make this more precise?

also would this be easier if we use arrays instead of sets? take a sorted array and test for deep equality, that should give a better error message automatically; what's the actual benefit of using a set here?

Reply:
Yea the strict equality of two sorted arrays covers us, if either side has items added or removed it should show up